### PR TITLE
Refactor bench.sh to make it more generic

### PR DESCRIPTION
**Pull request by harendra-kumar**
So that it can be reused in streaming-benchmarks package too.

Also add comparison option to compare the set of benchmarks provided on
the command line.

### CODE REVIEW

**Comment by harendra-kumar** [comment]
Better reviewed commit by commit.

File: bin/bench.sh [hunk 1]
```diff
@@ -300,16 +316,21 @@ run_reports() {
 
 cd $SCRIPT_DIR/..
 
-USE_GAUGE=0
 USE_GIT_CABAL=1
 USE_NIX=0
 set_common_vars
 
-DEFAULT_FIELDS="allocated bytescopied cputime"
-ALL_FIELDS="$FIELDS time cycles utime stime minflt majflt nvcsw nivcsw"
```
Comment (adithyaov) [inline bin/bench.sh:309]: Was this incorrect initially? `FIELDS` defined after `ALL_FIELDS`?

File: streamly.cabal [hunk 2]
```diff
@@ -62,6 +62,8 @@ extra-source-files:
     benchmark/README.md
     benchmark/bench-report/BenchReport.hs
     benchmark/bench-report/bench-report.cabal
+    benchmark/bench-report/bin/bench-runner.sh
+    benchmark/bench-report/bin/build-lib.sh
```
Comment (adithyaov) [inline streamly.cabal:66]: This should've ideally been in the previous commit. But, it's no big deal.

File: benchmark/bench-report/bin/bench-runner.sh [hunk 3]
```diff
@@ -77,47 +77,24 @@ list_comparisons ()  {
...
     hash -r
     if test ! -x "$prog_path" -a "$BUILD_ONCE" = "0"
     then
       echo "Building bench-report executables"
       BUILD_ONCE=1
-      pushd $BENCHMARK_DIR/bench-report
+      pushd $BENCH_REPORT_DIR
       local cmd
-      cmd="$CABAL_EXECUTABLE install --installdir $SCRIPT_DIR bench-report"
+      cmd="$CABAL_EXECUTABLE install --installdir bin bench-report"
```
Comment (adithyaov) [inline benchmark/bench-report/bin/bench-runner.sh:90]: `$BENCH_REPORT_DIR/bin` instead of `bin`. Use absolute path?

Comment (adithyaov) [reply benchmark/bench-report/bin/bench-runner.sh:90]: Nvm. It's fine for now.

File: bin/targets.sh [hunk 4]
```diff
@@ -0,0 +1,117 @@
+#------------------------------------------------------------------------------
```
Comment (adithyaov) [inline bin/targets.sh]: Ah! you've removed and re-introduced targets.

**Review by adithyaov — APPROVED** [review]
Minor comments. I don't see any obvious problems.

File: bin/bench.sh (hunk 1 above)
Comment (harendra-kumar) [reply bin/bench.sh:309]: yes.

**Pull request merged**